### ISSUE TITLE
PostgreSQL build wrong batch insert statement

### ISSUE THREAD

**Issue by mhashimmaqbool** [yiisoft/yii2#8172]
I'm getting following exception when I do batch insert:

################ Exception ##########################
Database Exception – yii\db\Exception
SQLSTATE[42601]: Syntax error: 7 ERROR: zero-length delimited identifier at or near """"
LINE 1: INSERT INTO "mobile_quiz_answers" ("", "", "", "") VALUES
^
The SQL being executed was: INSERT INTO "mobile_quiz_answers" ("", "", "", "") VALUES

Error Info: Array
(
    [0] => 42601
    [1] => 7
    [2] => ERROR:  zero-length delimited identifier at or near """"
LINE 1: INSERT INTO "mobile_quiz_answers" ("", "", "", "") VALUES                                            ^
)
#####################################################33

My code:

Yii::$app->db->createCommand()->batchInsert(Answer::tableName(), $modelAnswer->attributes, $rows)->execute();

And $row contain following data:

Array
(
    [0] => Array
        (
            [question_id] => 45493
            [answer] => a
            [correct] => 0
        )
    [1] => Array
        (
            [question_id] => 45493
            [answer] => b
            [correct] => 1
        )
..
..
)

**Comment by samdark** [comment]
Have you tried code from current master branch?

**Comment by mhashimmaqbool** [comment]
Thanks for the response.

I have resolved the error:

Actually in the documentation, it was mentioned the usage of batch insert in this way:

Yii::$app->db->createCommand()->batchInsert(<table name>, $model->attribtues, $rows)->execute();

But in my case $model->attributes was also returning 'id' as well, that was an auto-increment field. So, I won't need that. Also it was defined in my $rows array, that's why getting the errors. To resolve the problem, I just manually defined the attribute in an array like this:

Yii::$app->db->createCommand()->batchInsert(<table name>, ['attribute1', 'attribute2'], $rows)->execute();

And the problem has resolved.
